types/sap__approuter/index.d.ts

```diff
@@ -0,0 +1,428 @@
+         * @param sessionString a session object serialized to string
+         * @param timeout a timestamp in milliseconds, after which the session should be automatically invalidated
+         * @param callback  a function that is called after the session is saved in the internal session storage of the application router
+         */
+        set(sessionId: string, sessionString: string, timeout: number, callback: () => void): void;
+
+        /**
+         * @param sessionId an unsigned session identifier
+         * @param callback  function(currentSession) function, which returns session object. Callback function may modify and return current session object or create and return brand new session object
+         * currentSession - current session object
+         * @param resetTimeout a boolean that indicates whether to reset the session timeout
+         */
+        update(sessionId: string, callback: (currentSession: object) => void, resetTimeout: boolean): void;
+
+        /**
+         * @param sessionId an unsigned session identifier
+         * @param callback a function that is called after the session is destroyed in the internal session storage of the application router
+         */
+        destroy(sessionId: string, callback: () => void): void;
+
+        /**
+         * Emitted when a user session has been updated.
+         *
+         * @param event
+         */
+        on(event: "update", listener: (sessionId: string) => void): this;
+    }
+
+    type SessionStore = MemoryStore;
+
+    /**
+     * SAP Application Router
+     *
+     * For full help, please consult the README file in @sap/approuter (the source npm package) or
+     * https://help.sap.com/viewer/65de2977205c403bbc107264b8eccf4b/LATEST/en-US/050d87a61faa4fb88f687abd7bdf16ce.html
+     *
+     * These types (and their tests) are mostly extracted from the above mentioned README file and `doc/extending.md`
+     * in @sap/approuter (the source npm package). If some types are incorrect or missing, please open up an issue at
+     * {@link https://github.com/DefinitelyTyped/DefinitelyTyped}, or, even better, add them by yourself via a Pull Request.
+     *
+     * ## Session Management
+     * Session Management, as explained in `doc/sessionManagement.md` in @sap/approuter (the source npm package), is
+     * currently not typed. Feel free to help out with a Pull Request, if you want to or need these types.
+     *
+     * ## Command Parser
+     * The {@link cmdParser} property is set to a {@link Command} instance when instantiating an Approuter. It is possible
+     * to disable this feature by setting {@link cmdParser} to false. To do this typesafe, you have to use
+     * `Approuter<false>` as a type when creating the Approuter instance.
+     * @example
+     *   const ar = new Approuter<false>();
+     *   ar.cmdParser = false;
+     */
+    class Approuter<CommandParser extends Command | false = Command> extends EventEmitter {
```

Comment:
again no class exists on object at runtime on imported library. so this cannot be here or cannot be class

Reply:
changed it to be an interface, too